src/main/java/com/bytecoders/pharmaid/PharmaidController.java

```diff
@@ -153,6 +190,59 @@ public ResponseEntity<?> addPrescription(@PathVariable("userId") String userId,
     }
   }
 
+  /**
+   * Endpoint to update a user's prescriptions.
+   *
+   * @param userId user whose prescriptions we're trying to retrieve
+   * @param prescriptionId  prescription we are trying to update
+   * @param request containg prescription update
+   * @return a ResponseEntity with updated prescription if the operation is successful,
+   *      or an error message if an error occurred
+   */
+  @PatchMapping(path = "/users/{userId}/prescriptions/{prescriptionId}")
+  public ResponseEntity<?> updateUsersPrescription(
+      @PathVariable(USER_ID) String userId,
+      @PathVariable(PRESCRIPTION_ID) String prescriptionId,
+      @RequestBody @Valid UpdatePrescriptionRequest request) {
+    try {
+      final Optional<User> userOptional = userService.getUser(userId);
+
+      if (userOptional.isEmpty()) {
+        return new ResponseEntity<>("Provided User does not exist", HttpStatus.NOT_FOUND);
+      }
+
+      final Optional<Prescription> prescriptionOptional =
+            prescriptionService.getPrescriptionById(prescriptionId);
+
+      if (prescriptionOptional.isEmpty() 
+          || !prescriptionOptional.get().getUser().getId().equals(userId)) {
+        return new ResponseEntity<>(
+          "Provided Prescription doesn't exist",
+          HttpStatus.NOT_FOUND);
+      }
+
+      Prescription prescription = prescriptionService.getPrescriptionById(prescriptionId).get();
+
+      if (request.getEndDate() != null) {
+        prescription.setEndDate(request.getEndDate());
+      }
+      
+      if (request.getIsActive() != null) {
+        prescription.setIsActive(request.getIsActive());
+      }
```

Comment:
We need to change isActive depending on if endDate is null or not, or if currentDate > endDate.

I think the easiest way to do this is in `pharmaid/repository/model/Prescription.java` with:
```
  public void updateIsActive() {
      Date currentDate = new Date();
      this.isActive = (this.endDate != null && this.endDate.after(currentDate));
  }
```

Then, call `prescription.updateIsActive()` in addPrescription() function in Controller. Also need to remove it from `createPrescriptionRequest` and `updatePrescriptionRequest`. It should be removed from here too, but you should just call `prescription.updateIsActive()` again. 

We should have some tests for this to make sure we're catching this

Reply:
We should also have a validation check to ensure startDate <= endDate in `model/Prescription.java`:
```
@AssertTrue(message = "End date must be after or equal to start date if provided")
  public boolean isValidEndDate() {
      return endDate == null || !endDate.before(startDate);
  }
```